debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/launchConfigurations/LaunchConfigurationsDialog.java

```diff
@@ -1374,9 +1375,13 @@ public void run(boolean fork, boolean cancelable, IRunnableWithProgress runnable
 			if (cancelable) {
 				fProgressMonitorPart.attachToCancelComponent(null);
 			}
-			fProgressMonitorPart.getParent().setVisible(true);
+
 			fActiveRunningOperations++;
 
+			UIJob showProgressMonitorPart = createJobToShowProgressMonitorPart();
```

Comment:
```suggestion
					UIJob showProgressMonitorPart = UIJob.create(IInternalDebugCoreConstants.EMPTY_STRING, m -> {
			// only show the progress monitor if there are still active
			// operations by the time this job is executed
			if (fActiveRunningOperations > 0) {
				fProgressMonitorPart.getParent().setVisible(true);
			}
		});
```

Reply:
hm, I see your point but I'd rather have it in a separate method. I think the `run` method is a bit cluttered and a method with a descriptive name helps me "unload my brain" when I read through the code.

Reply:
OK, since it is your PR that's your decision. :)
But you can still use the mentioned factory in that method.

Reply:
Agreed, done in f50e586b30c24531456f1e19b348228a36044596